File: src/com/peyo/rplauncher/Launcher.java
package com.peyo.rplauncher;

import java.util.ArrayList;

import android.app.Activity;
import android.os.Bundle;

public class Launcher extends Activity implements LauncherModel.Callbacks {
    private LauncherModel mModel;
    private GridFragment mFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        if (savedInstanceState == null) {
            mFragment = new GridFragment();
            getFragmentManager().beginTransaction()
                    .replace(R.id.grid_fragment, mFragment)
                    .commit();
        }
        mModel = new LauncherModel(getApplicationContext(), this);
        mModel.startLoader();
    }

    @Override
    public void bindAllApplications(ArrayList<AppInfo> apps) {
        mFragment.updateApps(apps);
    }
}

File: src/com/peyo/rplauncher/Launcher.kt
package com.peyo.rplauncher

import android.app.Activity
import android.os.Bundle
import android.provider.Settings
import java.util.ArrayList

class Launcher : Activity(), LauncherModel.Callbacks {
    lateinit var mModel: LauncherModel
    lateinit var mFragment: GridFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (savedInstanceState == null) {
            mFragment = GridFragment()
            fragmentManager.beginTransaction()
                    .replace(R.id.grid_fragment, mFragment)
                    .commit()
        }
        mModel = LauncherModel(applicationContext, this)
        mModel.startLoader()
    }

    override fun bindAllApplications(apps: ArrayList<AppInfo>) {
        mFragment.updateApps(apps)

        Settings.Secure.putInt(getContentResolver(),
                Settings.Secure.TV_USER_SETUP_COMPLETE, 1)	
    }
}
